src/MPoly.jl

```diff
@@ -1271,69 +1297,57 @@ end
-argument `cached` to `false` will prevent the parent object `T` from being
-cached. `S` is a symbol corresponding to the ordering of the polynomial and
-can be one of `:lex`, `:deglex` or `:degrevlex`.
+    polynomial_ring(R::Ring, s::Vector{T}; cached::Bool = true, ordering::Symbol = :lex) where T <: Union{Symbol, AbstractString, Char}
+
+Given a base ring `R` and a vector `s` of variable names $x1, x2, \dots$ specifying
+how the generators (variables) should be printed, return a tuple `S, [x1, x2, ...]`
+representing the new polynomial ring $S = R[x1, x2, ...]$ and the generators
+$x1, x2, \dots$ of the polynomial ring.
+
+Mathematically the object `S` depends only on `R` and `x1, x2, ...` and by
+default it will be cached, i.e., if `polynomial_ring` is invoked again with the
+same arguments, the same (*identical*) ring is returned. Setting the optional
+argument `cached` to `false` ensures a distinct new ring is returned, and will
+also prevent it from being cached.
+
+The `ordering` of the polynomial ring can be one of `:lex`, `:deglex` or `:degrevlex`.
 """
-function polynomial_ring(R::AbstractAlgebra.Ring, s::Vector{String};
-                                   cached::Bool = true, ordering::Symbol = :lex)
-   return polynomial_ring(R, [Symbol(v) for v in s];
-                                   cached=cached, ordering=ordering)
-end
+polynomial_ring(R::Ring, s::Union{Tuple{Vararg{T}}, AbstractVector{T}}; kw...) where
+      T<:Union{Symbol, AbstractString, Char} =
+   polynomial_ring(R, Symbol[Symbol(x) for x in s]; kw...)
 
-function polynomial_ring(R::AbstractAlgebra.Ring, s::Vector{Char};
-                                   cached::Bool = true, ordering::Symbol = :lex)
-   return polynomial_ring(R, [Symbol(v) for v in s];
-                                   cached=cached, ordering=ordering)
+function polynomial_ring(R::Ring, s::Vector{Symbol}; kw...)
+   S = polynomial_ring_only(R, s; kw...)
+   (S, gens(S))
 end
 
-function polynomial_ring(R::AbstractAlgebra.Ring, s::Vector{Symbol};
-                                   cached::Bool = true, ordering::Symbol = :lex)
-   return Generic.polynomial_ring(R, s; cached=cached, ordering=ordering)
-end
+@doc md"""
+    polynomial_ring_only(R::Ring, s::Vector{Symbol}; ordering::Symbol=:lex, cached::Bool=true)
+
+Like [`polynomial_ring(R::Ring, s::Vector{Symbol})`](@ref) but return only the
+multivariate polynomial ring.
+"""
+polynomial_ring_only(R::T, s::Vector{Symbol}; ordering::Symbol=:lex, cached::Bool=true) where T<:Ring =
+   mpoly_ring_type(T)(R, s, ordering, cached)
+
+# Alternative constructors
 
 @doc Markdown.doc"""
-    polynomial_ring(R::Ring, n::Int, s::String; cached::Bool = false, ordering::Symbol = :lex)
+    polynomial_ring(R::Ring, n::Int, s::T = :x; cached, ordering) where T<:Union{Symbol, AbstractString, Char}
 
-Given a string `s` and a number of variables `n` will
+Given a symbol/string `s` and a number of variables `n` will
```

Comment:
Why leave out char? How about
```suggestion
Given a symbol, string or character `s` and a number of variables `n` will
```
Also "will automatically numbered is not valid english, I think (not your fault, of course). Maybe "... produce `n` variables with prefix `s` numbered from 1 to `n`. " ?